### PR TITLE
Jenkins pipeline

### PR DESCRIPTION
**Pull request by mabe13**
This pull request will add a Jenkins pipeline that can create an RDS instance in AWS and populate the database or it can be used to tear down (destroy) an RDS instance in AWS.

### CODE REVIEW

File: build/Jenkinsfile [hunk 1]
```diff
@@ -3,28 +3,53 @@ pipeline {
     label 'jenkins-slave'
   }
   options {
-    buildDiscarder(logRotator(numToKeepStr: '13'))
-    disableConcurrentBuilds()
+    buildDiscarder(logRotator(daysToKeepStr: '6'))
+  }
+  parameters {
+    string(
+        name: 'rds_instance_name',
```
Comment (JackPlowman) [inline build/Jenkinsfile:10]: What happens when a blank name is destroyed? or if a user enters a * can they delete many databases such as a user enters dos* could they delete all databases with a dos prefix?

File: Makefile [hunk 2]
```diff
@@ -53,9 +52,14 @@ push-dos-database-image: # Pushes the database image to the ECR repository
 	make docker-login
 	make docker-push NAME=data
 
-populate-database: # Deploys a kubernetes job to populate the RDS database
+populate-database: # Deploys a kubernetes job to populate the RDS database - mandatory: INSTANCE_NAME=[name];
+	export INSTANCE_NAME=$(INSTANCE_NAME) && \
```
Comment (stefaniuk) [inline Makefile:56]: Do we really need that line? Any make variable should become a shell variable anyway

File: Makefile [hunk 3]
```diff
@@ -53,9 +52,14 @@ push-dos-database-image: # Pushes the database image to the ECR repository
...
 	make k8s-deploy-job STACK=service PROFILE=non-prod
 
+delete-populate-database-k8s-deployment: # Delete kubernetes deployment for populate database job - mandatory: INSTANCE_NAME=[name];
+	export INSTANCE_NAME=$(INSTANCE_NAME) && \
```
Comment (stefaniuk) [inline Makefile:60]: Do we really need that line? Any make variable should become a shell variable anyway

File: build/Jenkinsfile [hunk 4]
```diff
@@ -3,28 +3,53 @@ pipeline {
...
+        sh 'make devops-print-variables'
+      }
+    }
+
+    stage('Create RDS Instance'){
+      when {
+        expression { params.build_action == 'create' }
+      }
+      steps {
+        sh 'make docker-config'
```
Comment (stefaniuk) [inline build/Jenkinsfile:31]: Can we extract `make docker-config` to a stage called `Prepare`?

File: build/Jenkinsfile [hunk 5]
```diff
@@ -3,28 +3,53 @@ pipeline {
...
-        script {
-          sh '''
-            export AWS_ACCOUNT_ID_LIVE_PARENT=000000000000
-            export AWS_ACCOUNT_ID_MGMT=000000000000
-            export AWS_ACCOUNT_ID_NONPROD=000000000000
-            export AWS_ACCOUNT_ID_PROD=000000000000
-            make devops-test-suite || make devops-test-suite DEBUG=1
-          '''
-        }
+        sh 'make docker-config'
```
Comment (stefaniuk) [inline build/Jenkinsfile:51]: Can we extract `make docker-config` to a stage called `Prepare`?

File: build/automation/var/profile/non-prod.mk [hunk 6]
```diff
@@ -4,7 +4,8 @@
 # Service variables
 
 DB_NAME=postgres?sslmode=require
-DB_HOST=test-dos-db-4-rds-nonprod.cqger35bxcwy.eu-west-2.rds.amazonaws.com
+#INSTANCE_NAME=
```
Comment (stefaniuk) [inline build/automation/var/profile/non-prod.mk:7]: Can we move `#INSTANCE_NAME=` above `DB_NAME=` please?

File: build/Jenkinsfile [hunk 7]
```diff
@@ -3,28 +3,53 @@ pipeline {
...
       }
     }
-    stage("Run Test Suite") {
+
+    stage('Populate Database'){
+      when {
+        expression { params.build_action == 'create' }
+      }
+      steps {
+        sh 'export DB_PASSWORD=$(make secret-fetch NAME=${rds_instance_name}-nonprod-dos_db_password) && make populate-database INSTANCE_NAME=${rds_instance_name}'
```
Comment (stefaniuk) [inline build/Jenkinsfile:41]: This is unnecessary complicated. Please, can we wrap it in a single make target?

**Pull request merged**